ecs-agent/tcs/handler/handler_test.go

```diff
@@ -340,32 +352,103 @@ func TestConnectionInactiveTimeout(t *testing.T) {
 		testAgentVersion,
 		testAgentHash,
 		testContainerRuntimeVersion,
-		server.URL,
 		false,
 		testCreds,
 		testCfg,
 		deregisterInstanceEventStream,
-		50*time.Millisecond,
+		5*time.Second,
 		100*time.Millisecond,
 		testDisconnectionTimeout,
 		testDisconnectionJitter,
 		nil,
 		telemetryMessages,
 		healthMessages,
 		emptyDoctor,
+		testecsclient,
 	)
 
 	// Start a session with the test server.
-	err = session.StartTelemetrySession(ctx, server.URL)
+	err = session.StartTelemetrySession(ctx)
 
-	assert.NoError(t, err, "Close the connection should cause the tcs client return error")
+	// TODO: confirm once. The client would not attempt to reconnect since we are using StartTelemetrySession
+	// instead of Start. Modified the test to just test for connection closed on inactivity.
+	//assert.NoError(t, err, "Close the connection should cause the tcs client return error")
+	msg := <-serverErr
 
-	assert.True(t, websocket.IsCloseError(<-serverErr, websocket.CloseAbnormalClosure),
+	assert.True(t, websocket.IsCloseError(msg, websocket.CloseAbnormalClosure),
 		"Read from closed connection should produce an io.EOF error")
 
 	closeSocket(closeWS)
 }
 
+// TestClientReconnectsAfterInactiveTimeout tests the tcs client reconnects when it loses network
+// connection, or it's inactive for too long
+func TestClientReconnectsAfterInactiveTimeout(t *testing.T) {
+	// Start test server.
+	closeWS := make(chan []byte)
+	server, _, requestChan, _, err := wsmock.GetMockServer(closeWS)
+	server.StartTLS()
+	defer server.Close()
+	if err != nil {
```

Comment:
Shouldn't this check be next to line # 389

Reply:
Yes, you are right. Thank you for pointing it out. Will fix it.